clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -8267,6 +8267,10 @@ void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
     // Symbol file and specialization constant info generation is mandatory -
     // add options unconditionally
     addArgs(CmdArgs, TCArgs, {"-symbols"});
+    // By default we split SYCL and ESIMD kernels into separate modules
+    if (TCArgs.hasFlag(options::OPT_fsycl_split_esimd,
+                       options::OPT_fno_sycl_split_esimd, true))
+      addArgs(CmdArgs, TCArgs, {"-split-esimd"});
```

Comment:
In some cases Driver adds `-ir-output-only` to post link invocation, AFAIR. This will conflict with -split-esimd then when there are both SYCL and ESIMD kernels. What error diagnostics user will see in that case?

Reply:
Right now the driver will never call `sycl-post-link` with both `-ir-output-only` and `-split-esimd` options.

By default, the driver will call `sycl-post-link` with `-split-esimd`.
If a user calls `clang++ -fintelfpga`, then the driver calls `sycl-post-link` with `-ir-output-only`.